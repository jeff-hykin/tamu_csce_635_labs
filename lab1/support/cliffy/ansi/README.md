# Ansi

<p align="center" class="badges-container">
  <a href="https://github.com/c4spar/deno-cliffy/releases">
    <img alt="Version" src="https://img.shields.io/github/v/release/c4spar/deno-cliffy?logo=github&color=blue" />
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/actions/workflows/test.yml">
    <img alt="Build status" src="https://github.com/c4spar/deno-cliffy/workflows/Test/badge.svg?branch=main" />
  </a>
  <a href="https://codecov.io/gh/c4spar/deno-cliffy">
    <img src="https://codecov.io/gh/c4spar/deno-cliffy/branch/main/graph/badge.svg"/>
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/labels/module%3Aansi-escape">
    <img alt="issues" src="https://img.shields.io/github/issues/c4spar/deno-cliffy/module:ansi-escape?label=issues&logo=github&color=yellow">
  </a>
  <a href="https://deno.land/">
    <img alt="Deno version" src="https://img.shields.io/badge/deno-^1.4.0-blue?logo=deno" />
  </a>
  <a href="https://doc.deno.land/https/deno.land/x/cliffy/ansi/mod.ts">
    <img alt="doc" src="https://img.shields.io/badge/deno-doc-yellow?logo=deno" />
  </a>
  <a href="https://discord.gg/ghFYyP53jb">
    <img alt="Discord" src="https://img.shields.io/badge/join-chat-blue?logo=discord&logoColor=white" />
  </a>
  <a href="../LICENSE">
    <img alt="License" src="https://img.shields.io/github/license/c4spar/deno-cliffy?logo=github" />
  </a>
  <br>
  <a href="https://deno.land/x/cliffy">
    <img alt="Discord" src="https://img.shields.io/badge/Published on deno.land-blue?logo=deno&logoColor=959DA6&color=272727" />
  </a>
</p>

The documentation has been moved to https://cliffy.io/docs/ansi.
