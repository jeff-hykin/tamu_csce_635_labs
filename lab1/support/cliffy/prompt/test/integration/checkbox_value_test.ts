import { ansi } from "../../../ansi/ansi.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";
import { Checkbox } from "../../checkbox.ts";

await snapshotTest({
  name: "[checkbox] should handle none primitive values",
  meta: import.meta,
  osSuffix: ["windows"],
  stdin: ansi
    .cursorDown
    .text(" ")
    .text("\n")
    .text("\n")
    .toArray(),
  async fn() {
    await Checkbox.prompt({
      message: "Select an option",
      options: [
        { name: "date 1", value: new Date(1000) },
        { name: "date 2", value: new Date(2000) },
      ],
    });
  },
});
