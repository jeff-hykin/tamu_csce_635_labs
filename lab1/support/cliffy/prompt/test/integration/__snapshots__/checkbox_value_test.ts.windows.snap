export const snapshot = {};

snapshot[`[checkbox] should handle none primitive values 1`] = `
stdout:
"? Select an option
❯ × date 1
  × date 2\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  × date 1
❯ × date 2\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  × date 1
❯ √ date 2\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  × date 1
❯ √ date 2
Press ↵ again to submit. To check or uncheck the selected option press space.\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option » date 2
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
